keepalived/check/check_tcp.c

```diff
@@ -62,6 +62,24 @@ dump_tcp_check(void *data)
 	}
 }
 
+static int
+compare_tcp_check(void *a, void *b)
+{
+	tcp_check_t *old = CHECKER_DATA(a);
+	tcp_check_t *new = CHECKER_DATA(b);
+
+	if (compare_conn_opts(CHECKER_CO(a), CHECKER_CO(b)) != 0)
+		goto err;
+	if (old->n_retry != new->n_retry)
+		goto err;
+	if (old->delay_before_retry != new->delay_before_retry)
+		goto err;
```

Comment:
As delay above, I don't think different values of n_retry or delay_before_retry make this a different checker.